pkg/ottl/ottlfuncs/func_replace_pattern_test.go

```diff
@@ -15,13 +16,42 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
 )
 
+type optionalFnTestArgs[K any] struct {
+	Target ottl.StringGetter[K]
+}
+
+func NewTestFactory[K any]() ottl.Factory[K] {
```

Comment:
This function should be unexported and be given a more specific name as well.

Reply:
called it `optionalFnTestFactory`